src/js/capture.js

```diff
@@ -160,15 +175,21 @@ const capture = {
     } else {
       firstPartyUrl = new URL(response.originUrl);
     }
-
-    if (firstPartyUrl.hostname
-      && targetUrl.hostname !== firstPartyUrl.hostname
-      && await this.shouldStore(response)) {
+    //  Trying to stop all the loops of calls to the API!!
+    const greenStatus = await checkGreenStatus(targetUrl.hostname).then(data => {
```

Comment:
Because you're using await here, I don't think you'd need to use the `then` afterwards to fetch the data from resolved promise that `checkGreenStatus` returns as the `await` stuff is mainly there to avoid needing faff around with promises.

You should be able to do this:

```
const greenStatus = await checkGreenStatus(documentUrl.hostname)
const greenCheckData = greenStatus.data
```

Then  further down, you'd do this instead of needing the `then` function where you were returning data.green before

```
const data = {
        faviconUrl: tab.favIconUrl,
        firstParty: true,
        requestTime: Date.now(),
        greenCheck: greenCheckData.green
      };

Reply:
Yup, that makes a lot of sense. Thank you!